packages/nextalign/src/translate/mapCoordinates.cpp

```diff
@@ -27,8 +27,12 @@ namespace {
     int refPos = 0;
     for (int i = 0; i < alnLength; ++i) {
       if (ref[i] == Nucleotide::GAP) {
-        const auto& prev = revCoordMap.back();
-        revCoordMap.push_back(prev);
+        if (revCoordMap.empty()) {
+          revCoordMap.push_back(0);
+        } else {
+          const auto& prev = revCoordMap.back();
+          revCoordMap.push_back(prev);
+        }
       } else {
```

Comment:
The access to the last item with `std::vector::back()` of the potentially empty vector was unguarded previously:
```cpp
const auto& prev = revCoordMap.back();
```

I added a conditional to push `0` if the array is empty. Tests pass cleanly, so I think it does not change the behavior, except that the first few elements now contain a well-defined value `0` instead of random garbage from heap.